sdk/identity/identity/migration-v1-v2.md

```diff
@@ -0,0 +1,307 @@
+# Migrate from v1 to v2 of @azure/identity
+
+Version 2 of the `@azure/identity` package includes the best parts of version 1, plus several improvements.
+
+If you're moving from version 1 to version 2, you're in the right place. In this document we will show you what needs to be updated.
+
+For the full set of changes and bug fixes, please refer to our changelog: https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/identity/identity/CHANGELOG.md#200-2021-10-12
+
+## Table of contents
+
+- [Update your package.json](#update-your-package-json)
+- [Update the handling of authentication errors](#update-the-handling-of-authentication-errors).
+- [Controlling interactive authentication](#controlling-interactive-authentication).
+- [Changes to your browser authentication flow](#changes-to-your-browser-authentication-flow).
+- [Asynchronous verification of certificates on ClientCertificateCredential](#asynchronous-verification-of-certificates-on-clientcertificatecredential).
+- [Other changes to existing credentials](#other-changes-to-existing-credentials).
+- [New Plugins API](#new-plugins-api)
+  - [Persistent token cache plugin](#persistent-token-cache-plugin)
+  - [VisualStudioCodeCredential plugin](#visualstudiocodecredential-plugin)
+- [Troubleshooting](#troubleshooting)
+- [Provide feedback](#provide-feedback)
+- [Contributing](#contributing)
+
+## Update your package.json
+
+First, make sure to update your package.json to point to the version 2 of Identity:
+
+```diff
+  "dependencies": {
+-     "@azure/identity": "^1.5.2",
++     "@azure/identity": "^2.0.0"
+  }
+```
+
+If you're using the `VisualStudioCodeCredential`, you likely have `keytar` in your dependencies. Our version 2 of Identity does not have any binary dependency. We have moved these to new Plugin packages. To continue using the `VisualStudioCodeCredential`, please install the [`@azure/identity-vscode`](https://www.npmjs.com/package/@azure/identity-vscode) package.
```

Comment:
`Update your package.json` section is something I would expect most customers to skip as they would know how to update their dependencies. They will miss out on this part where you are talking about VS Code credential.

I would recommend either removing the parts about updating package.json and renaming this section to reflect the changes to VS Code credential or add a separate section for VS Code credential

Reply:
I will do a significant update that might make you re-consider this. Give me some time to circle through these documents with Scott A. I’ll get back to you asap.